src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -60,20 +70,32 @@
     private final DialogService dialogService;
     private final BibEntry entry;
     private final TaskExecutor taskExecutor;
+    private final FileDirectoryPreferences fileDirectoryPreferences;
+    private final CleanupPreferences cleanupPreferences;
+    private final LayoutFormatterPreferences layoutFormatterPreferences;
+    private final XmpPreferences xmpPreferences;
+    private final String fileNamePattern;
 
+    @Deprecated
     public LinkedFileViewModel(LinkedFile linkedFile, BibEntry entry, BibDatabaseContext databaseContext, TaskExecutor taskExecutor) {
-        this(linkedFile, entry, databaseContext, taskExecutor, new FXDialogService());
+        this(linkedFile, entry, databaseContext, taskExecutor, new FXDialogService(), Globals.prefs, Globals.journalAbbreviationLoader);
     }
 
-    protected LinkedFileViewModel(LinkedFile linkedFile, BibEntry entry, BibDatabaseContext databaseContext,
-                                  TaskExecutor taskExecutor, DialogService dialogService) {
+    public LinkedFileViewModel(LinkedFile linkedFile, BibEntry entry, BibDatabaseContext databaseContext,
```

Comment:
I think, only the `IMPORT_FILENAMEPATTERN` and `preferences.getFileDirectoryPreferences()` are needed. Why not pass them both (as quick fix regarding the information hiding). OK, a new preference object would even be better, but that costs more time.

Reply:
Also the auto-link preferences as well as the import preferences are needed. You are definitely right, a custom preference object unifying them all would be nice. To be honest, I don't have the motivation right now to create one (especially since the focus of this PR is a different one).